### ISSUE TITLE
Incorrect Form usage guide example/preview

### ISSUE THREAD

**Issue by Hackerokuz** [huntabyte/shadcn-svelte#764]
### Change Type

Correction

### Proposed Changes

Usage section has to be updated as currently it is not correct. Current setup causes issues with typescript.

# Usage

1.  For `src/routes/settings/settings-form.svelte` we need to use zod adapter from `sveltekit-superforms/adapters` for `formSchema`

```typescript
import type { PageServerLoad, Actions } from "./$types";
import { fail } from "@sveltejs/kit";
import { superValidate } from "sveltekit-superforms/server";
import { zod } from "sveltekit-superforms/adapters";
import { formSchema } from "./schema";

const adaptedFormSchema = zod(formSchema);

export const load: PageServerLoad = async () => {
  return {
    form: await superValidate(adaptedFormSchema),
  };
};
export const actions: Actions = {
  default: async (event) => {
    const form = await superValidate(event, adaptedFormSchema);
    if (!form.valid) {
      return fail(400, {
        form,
      });
    }
    return {
      form,
    };
  },
};

```

2. In `src/routes/settings/schema.ts` we need to update `FormSchema` type to use zod infer type instead of straight typeof.

```typescript
import { z } from "zod";
export const formSchema = z.object({
    username: z.string().min(2).max(50),
});
export type FormSchema = z.infer<typeof formSchema>;
```

4. In `src/routes/settings/settings-form.svelte` we need to add `controlled` parameter to `Form.Root`
5. Then we need to create `superFrm` const to pass to `form` parameter to `Form.Root` using `superForm` function from `sveltekit-superforms/client`

```svelte
<script lang="ts">
	import * as Form from '$lib/components/ui/form';
	import { formSchema, type FormSchema } from './schema';
	import type { SuperValidated } from 'sveltekit-superforms';
	import { superForm } from 'sveltekit-superforms/client';
	import { zod } from 'sveltekit-superforms/adapters';
	export let form: SuperValidated<FormSchema>;
	const superFrm = superForm(form, { validators: zod(formSchema) });
</script>

<Form.Root method="POST" form={superFrm} schema={formSchema} let:config controlled>
	<Form.Field {config} name="username">
		<Form.Item>
			<Form.Label>Username</Form.Label>
			<Form.Input />
			<Form.Description>This is your public display name.</Form.Description>
			<Form.Validation />
		</Form.Item>
	</Form.Field>
	<Form.Button>Submit</Form.Button>
</Form.Root>

```

**Comment by huntabyte** [comment]
Formsnap hasn't been updated to support Supeforms v2 and this guide will be updated when that happens.

**Comment by kelsiesmurphy** [comment]
This open issue fixed what has been bugging me for days, hope Formsnap updates soon so the docs can show the correct code, and nobody else has to go through my pain! 😛

**Comment by HubbeDev** [comment]
@kelsiesmurphy well @huntabyte is doing a heck of a job over at [formsnap#131](https://github.com/huntabyte/formsnap/pull/131). Give it the time it needs and you will not be disappointed.

**Comment by huntabyte** [comment]
Closed by #776
